docs/rfc/rfc-19-random-beacon-v2.adoc

```diff
@@ -0,0 +1,225 @@
+
+[[fees-and-rewards]]
+=== Fees and rewards
+
+In tBTC v1, operators have to execute a lot of on-chain transactions per
+deposit. Even though they are not reimbursed for on-chain actions and they
+receive KEEP rewards and TBTC fees, this scheme proved to be successful and no
+operators complained about it.
+
+For v2 of the random beacon, all rewards should be done in T.
+This scheme is compatible with tBTC v2 design where all rewards and
+are also done only in T.
+
+Relay requester should provide a fee in T. The value of the fee is a governable
+parameter. The entire fee is deposited in the maintenance pool that is used to
+reimburse for different actions related to DKG.
+
+The transaction submitting relay entry is not reimbursable and implementation
+should ensure the gas cost of this transaction is as low as possible, below
+200k gas. The first group member eligible to submit the result is 
+`previous_entry MOD group_size`, then also `(previous_entry MOD group_size) + 1`.
+At the end, all group members should be eligible to submit the result. If the
+given member did not submit the result in their round, they should be removed
+from the sortition pool for at least 2 weeks and do not earn rewards for
+that time. The operation removing members from the sortition pool should be
+as cheap as possible so that the member submitting relay entry does not have to
+pay any additional costs for removing inactive members from the pool. Smart
+contract patterns such as gas station could be helpful in achieving this goal.
+
+For example, if `previous_entry MOD group_size = 60` and the entry was submitted 
+by the member with index `3`, members with indexes `{60, 61, 62, 63, 64, 1, 2}`
+should be removed from the sortition pool in the same transaction in which
+member with index `3` submitted relay entry.
+
+The default time for a single member to submit a relay entry should be increased 
+from 6 to 10 blocks and should become a governable parameter.
+
+There is a fixed, governable reward for submitting a DKG result, paid from the
+maintenance pool. The reward is paid to the DKG result submitter in the transaction
+approving the DKG result. The logic triggering new group selection will be embedded 
+in relay entry request transaction and should be as cheap as possible - it only
+locks the sortition pool and emits a group creation start event.
+
+In case the DKG result has not been submitted on time, anyone can unlock the pool
+and receive a fixed, governable reward for reporting DKG timeout. The reward is
+paid from the maintenance pool.
+
+Just like in the case of relay entry submission, the order in which operators
+are supposed to submit DKG result is enforced on-chain and operators who missed
+their round are removed from the sortition pool for at least 2 weeks and do
+not earn rewards for that time. The first member eligible to submit the DKG result
+is a member with index `1`, then also `2`. For example, if member `9` submitted
```

Comment:
we could also use `hash(public_key) % group_size` for unpredictability

Reply:
Agreed, it's better.

Reply:
Please see https://github.com/keep-network/keep-core/pull/2585/commits/5ca9187cf42019592e814451eb2d8935c9d41dde.